core/Archive/ArchiveInvalidator.php

```diff
@@ -52,35 +52,55 @@ class ArchiveInvalidator
      * @var Model
      */
     private $model;
+    /**
+     * Queue of archives to invalidate
+     */
+    private $archives_to_invalidate;
+
 
     public function __construct(Model $model)
     {
         $this->model = $model;
+        $this->archives_to_invalidate = array();
+    }
+
+    public function __destruct()
```

Comment:
If we want something to be run at the end of the request, would probably want to use `register_shutdown_function` than depend on `__destruct`. Actually may want to use the `'Tracker.end'` since this will be called before the DB connection is closed.

Reply:
That might explain why currently my patch doesn't actually write in the db (although there are no failures).